src/bun.js/http.exports.js

```diff
@@ -292,6 +486,209 @@ export class IncomingMessage extends Readable {
   }
 }
 
+function emitErrorNt(msg, err, callback) {
+  callback(err);
+  if (typeof msg.emit === "function" && !msg._closed) {
+    msg.emit("error", err);
+  }
+}
+
+function onError(self, err, cb) {
+  process.nextTick(() => emitErrorNt(self, err, cb));
+}
+
+function isUint8Array(arr) {
+  return Object.prototype.toString.call(arr) === "[object Uint8Array]";
+}
+
+function write_(msg, chunk, encoding, callback, fromEnd) {
+  if (typeof callback !== "function") callback = nop;
+
+  let len;
+  if (chunk === null) {
+    // throw new ERR_STREAM_NULL_VALUES();
+    throw new Error("ERR_STREAM_NULL_VALUES");
+  } else if (typeof chunk === "string") {
+    len = Buffer.byteLength(chunk, encoding);
+  } else if (isUint8Array(chunk)) {
+    len = chunk.length;
+  } else {
+    throw new Error("Invalid arg type for chunk");
+    // throw new ERR_INVALID_ARG_TYPE(
+    //   "chunk",
+    //   ["string", "Buffer", "Uint8Array"],
+    //   chunk,
+    // );
+  }
+
+  let err;
+  if (msg.finished) {
+    // err = new ERR_STREAM_WRITE_AFTER_END();
+    err = new Error("ERR_STREAM_WRITE_AFTER_END");
+  } else if (msg.destroyed) {
+    // err = new ERR_STREAM_DESTROYED("write");
+    err = new Error("ERR_STREAM_DESTROYED");
+  }
+
+  if (err) {
+    if (!msg.destroyed) {
+      onError(msg, err, callback);
+    } else {
+      process.nextTick(callback, err);
+    }
+    return false;
+  }
+
+  if (!msg._header) {
+    if (fromEnd) {
+      msg._contentLength = len;
+    }
+    // msg._implicitHeader();
+  }
+
+  if (!msg._hasBody) {
+    debug(
+      "This type of response MUST NOT have a body. " +
+        "Ignoring write() calls.",
+    );
+    process.nextTick(callback);
+    return true;
+  }
+
+  // if (!fromEnd && msg.socket && !msg.socket.writableCorked) {
+  //   msg.socket.cork();
+  //   process.nextTick(connectionCorkNT, msg.socket);
+  // }
+
+  return true;
+}
+
+const kClearTimeout = Symbol("kClearTimeout");
+
+export class OutgoingMessage extends Writable {
+  #headers = new Headers();
+  headersSent = false;
+  sendDate = true;
+  req;
+
+  #finished = false;
+  [kEndCalled] = false;
+
+  #socket;
+  #timeoutTimer = null;
+
+  // So that headers are accessible outside of `OutgoingMessage`
+  get _headers() {
+    return this.#headers;
```

Comment:
Are you sure it's supposed to return a Headers object? If this is an internal-only symbol ("protected") then can you make it a Symbol so that it doesn't appear as enumerable

Reply:
Yeah I think I was referencing the partial implementation from #1627, I don't think this is actually being used now. Removed it